### PR TITLE
Documentation: labs: device_drivers: update exercises

### PR DESCRIPTION
**Pull request by valighita**
Signed-off-by: Valentin Ghita valx92@gmail.com

### CODE REVIEW

File: Documentation/teaching/labs/device_drivers.rst [hunk 1]
```diff
@@ -948,46 +948,82 @@ For this:
...
+
+Add two ioctl operations to modify the message associated with the
+driver. Use fixed-length buffer ( BUFFER_SIZE ).
+
+  1. Add the ``ioctl`` function from the driver the following operations:
+     * ``MY_IOCTL_SET_BUFFER`` for writing a message to the device;
+     * ``MY_IOCTL_GET_BUFFER`` to read a message from your device.
+  2. Change the user-space program to allow testing.
+
+.. note:: Read the ioctl sections and Access to the address space of the lab process.
```
Comment (teoserbanescu) [inline Documentation/teaching/labs/device_drivers.rst]: Don't forget to add hyperlinks to the sections.

File: Documentation/teaching/labs/device_drivers.rst [hunk 2]
```diff
@@ -948,46 +948,82 @@ For this:
...
-.. Extra
-   -----
+When the process is added to the queue, it will remain blocked in execution; To
+run the queue command open a new console in the virtual machine with Alt+F2 ;
+You can return to the previous console with Alt+F1 . If you're connected via
+SSH to the virtual machine, open a new console.
 
-   Ioctl with messaging
-   --------------------
+.. note:: Read the ioctl and Synchronization sections - waiting queues in the lab.
```
Comment (teoserbanescu) [inline Documentation/teaching/labs/device_drivers.rst]: Don't forget to add hyperlinks to the sections.

File: tools/labs/templates/device_drivers/kernel/so2_cdev.c [hunk 3]
```diff
@@ -173,11 +179,14 @@ so2_cdev_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
...
 	.owner = THIS_MODULE,
-/* TODO 2/4: add open, release, read, write functions */
+/* TODO 2/2: add open and release functions */
 	.open = so2_cdev_open,
 	.release = so2_cdev_release,
+/* TODO 4/1: add read function */
 	.read = so2_cdev_read,
+/* TODO 5/1: add write function */
 	.write = so2_cdev_write,
+/* TODO 6/1: add write function */
```
[the remark this answers is not in the document]
Comment (valighita) [reply tools/labs/templates/device_drivers/kernel/so2_cdev.c]: Fixed

File: Documentation/teaching/labs/device_drivers.rst (hunk 1 above)
Comment (valighita) [reply Documentation/teaching/labs/device_drivers.rst]: Fixed.

File: Documentation/teaching/labs/device_drivers.rst (hunk 2 above)
Comment (valighita) [reply Documentation/teaching/labs/device_drivers.rst]: Fixed.

**Pull request merged**